app/mempool/README.md

```diff
@@ -0,0 +1,154 @@
+
+## Architecture
+
+### Core Components
+
+1. **Mempool**: The main structure that manages multiple transaction lanes and implements the core mempool interface.
+2. **Lane**: A logical grouping of transactions with specific matching criteria and space allocation.
+3. **Proposal**: Represents a block proposal under construction, managing transaction inclusion and space limits.
+4. **BlockSpace**: Manages block space constraints including transaction bytes and gas limits.
+
+### Key Features
+
+- **Lane-based Organization**: Transactions are organized into different lanes based on their matching functions
+- **Space Management**: Efficient management of block space and gas limits
+- **Transaction Prioritization**: Support for different transaction priorities among and within lanes 
+- **Thread Safety**: Built-in mutex protection for concurrent access
+- **Error Recovery**: Panic recovery mechanisms for robust operation
+
+## Usage
+
+### Creating a Lane
+
+A lane is created with specific matching criteria and space allocation. Here's an example of creating a lane for bank send transactions:
+
+```go
+bankSendLane := NewLane(
+    logger,
+    txEncoder,
+    signerExtractor,
+    "bankSend",                    // Lane name
+    isBankSendTx,                  // Matching function
+    math.LegacyMustNewDecFromStr("0.2"),  // Max transaction space ratio
+    math.LegacyMustNewDecFromStr("0.3"),  // Max lane space ratio
+    sdkmempool.DefaultPriorityMempool(),  // Underlying mempool implementation
+    nil,                           // Lane limit check handler
+)
+
+// Example matching function for bank send transactions
+func isBankSendTx(_ sdk.Context, tx sdk.Tx) bool {
+    msgs := tx.GetMsgs()
+    if len(msgs) == 0 {
+        return false
+    }
+    for _, msg := range msgs {
+        if _, ok := msg.(*banktypes.MsgSend); !ok {
+            return false
+        }
+    }
+    return true
+}
+```
+
+Key parameters for lane creation:
+- `logger`: Logger instance for lane operations
+- `txEncoder`: Function to encode transactions
+- `signerExtractor`: Adapter to extract signer information
+- `name`: Unique identifier for the lane
+- `matchFn`: Function to determine if a transaction belongs in this lane
+- `maxTransactionSpace`: Maximum space ratio for individual transactions (relative to total block space) more details on [Space management](#space-management) 
+- `maxLaneSpace`: Maximum space ratio for the entire lane (relative to total block space) more details on [Space management](#space-management)
+- `laneMempool`: Underlying Cosmos SDK mempool implementation that handles transaction storage and ordering within the lane. This determines how transactions are stored and selected within the lane
+- `handleLaneLimitCheck`: Optional callback function that is called when the lane exceeds its space limit. This can be used to implement inter-lane dependencies, such as blocking other lanes when a lane exceeds its limit.
```

Comment:
recheck a bit. signer extractor is already removed?

Reply:
will be rechecked all later